src/wp-includes/blocks.php

```diff
@@ -500,6 +579,43 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 		}
 	}
 
+	$module_fields = array(
+		'editorModule' => 'editor_module_handles',
+		'module'       => 'module_handles',
+		'viewModule'   => 'view_module_handles',
+	);
+	foreach ( $module_fields as $metadata_field_name => $settings_field_name ) {
+
+		if ( ! empty( $settings[ $metadata_field_name ] ) ) {
+			$metadata[ $metadata_field_name ] = $settings[ $metadata_field_name ];
+		}
+		if ( ! empty( $metadata[ $metadata_field_name ] ) ) {
+			$modules           = $metadata[ $metadata_field_name ];
+			$processed_modules = array();
+			if ( is_array( $modules ) ) {
+				for ( $index = 0; $index < count( $modules ); $index++ ) {
+					$result = register_block_module_handle(
+						$metadata,
+						$metadata_field_name,
+						$index
+					);
+					if ( $result ) {
+						$processed_modules[] = $result;
+					}
+				}
+			} else {
+				$result = register_block_module_handle(
+					$metadata,
+					$metadata_field_name
+				);
+				if ( $result ) {
+					$processed_modules[] = $result;
+				}
```

Comment:
Nothing is returned from register, just add the module ID to `$processed_modules`.

See https://github.com/WordPress/wordpress-develop/pull/5818#issuecomment-1871048068